packages/dev/core/src/Misc/sceneSerializer.ts

```diff
@@ -40,62 +43,74 @@ const SerializeMesh = (mesh: Mesh, serializationScene: any): any => {
     return serializationObject;
 };
 
-const FinalizeSingleMesh = (mesh: Mesh, serializationObject: any) => {
-    //only works if the mesh is already loaded
-    if (mesh.delayLoadState === Constants.DELAYLOADSTATE_LOADED || mesh.delayLoadState === Constants.DELAYLOADSTATE_NONE) {
-        const serializeMaterial = (material: Material) => {
-            serializationObject.materials = serializationObject.materials || [];
-            if (mesh.material && !serializationObject.materials.some((mat: Material) => mat.id === (<Material>mesh.material).id)) {
-                serializationObject.materials.push(material.serialize());
-            }
-        };
-
-        //serialize material
-        if (mesh.material && !mesh.material.doNotSerialize) {
-            if (mesh.material instanceof MultiMaterial) {
-                serializationObject.multiMaterials = serializationObject.multiMaterials || [];
-                if (!serializationObject.multiMaterials.some((mat: Material) => mat.id === (<Material>mesh.material).id)) {
-                    serializationObject.multiMaterials.push(mesh.material.serialize());
-                    for (const submaterial of mesh.material.subMaterials) {
-                        if (submaterial) {
-                            serializeMaterial(submaterial);
+const FinalizeSingleNode = (node: Node, serializationObject: any) => {
+    if (node.getClassName() === "Mesh" || node.getClassName() === "GroundMesh") {

```

Comment:
Just wondering if we are not losing Lines here ?

Reply:
And `GreasedLine` too, now.

So, maybe `if ((node as Mesh)._isMesh) {`?

Reply:
> So, maybe `if ((node as Mesh)._isMesh) {`?

@Popov72 This will also apply to TrailMesh, do we want that?

Reply:
Looking into the TrailMesh, I found out that whenever we serialize a TrailMesh and load it again, it "loses" it class, so we can't start it: https://playground.babylonjs.com/#40E17W#3 
This counts as a bug, right?

Reply:
Yes, I guess it's a bug!

Also, I think we do want `TrailMesh` to be serialized, as it extends `Mesh`, there's no reason why it should not work (except for the bug you found!).

Reply:
All rightie, adding a fix to this too 😄